src/platform/silabs/platformAbstraction/SilabsPlatformBase.h

```diff
@@ -56,6 +56,9 @@ class SilabsPlatformAbstractionBase
     virtual CHIP_ERROR FlashErasePage(uint32_t addr) { return CHIP_ERROR_NOT_IMPLEMENTED; }
     virtual CHIP_ERROR FlashWritePage(uint32_t addr, const uint8_t * data, size_t size) { return CHIP_ERROR_NOT_IMPLEMENTED; }
 
+    // soft reset
+    virtual void SoftwareReset(void) {}
```

Comment:
This can be pure virtual since it is a required function.

Reply:
Added as pure virtual function and comments.